lib/helpers/utils.js

```diff
@@ -1770,86 +1782,152 @@ export async function parsePnpmLock(pnpmLock, parentComponent = null) {
     }
     // This logic matches the pnpm list command to include only direct dependencies
     if (ppurl !== "") {
-      // In lock file version 9, direct dependencies is under importers
-      const rootDirectDeps =
-        lockfileVersion >= 9
-          ? yamlObj.importers["."]?.dependencies || {}
-          : yamlObj.dependencies || {};
-      const rootDevDeps =
-        lockfileVersion >= 9
-          ? yamlObj.importers["."]?.devDependencies || {}
-          : {};
-      const rootOptionalDeps =
-        lockfileVersion >= 9
-          ? yamlObj.importers["."]?.optionalDependencies || {}
-          : {};
-      const ddeplist = [];
-      // Find the root optional dependencies
-      for (const rdk of Object.keys(rootDevDeps)) {
-        let version = rootDevDeps[rdk];
-        if (typeof version === "object" && version.version) {
-          version = version.version;
-        }
-        // version: 3.0.1(ajv@8.14.0)
-        if (version?.includes("(")) {
-          version = version.split("(")[0];
-        }
-        const dpurl = new PackageURL(
-          "npm",
-          "",
-          rdk,
-          version,
-          null,
-          null,
-        ).toString();
-        possibleOptionalDeps[decodeURIComponent(dpurl)] = true;
-      }
-      for (const rdk of Object.keys(rootOptionalDeps)) {
-        let version = rootOptionalDeps[rdk];
-        if (typeof version === "object" && version.version) {
-          version = version.version;
-        }
-        // version: 3.0.1(ajv@8.14.0)
-        if (version?.includes("(")) {
-          version = version.split("(")[0];
+      if (yamlObj["importers"]) {
+        // In lock file version 9, direct dependencies is under importers
+        const rootDirectDeps =
+          lockfileVersion >= 9
```

Comment:
Can we make these `>= 6` in all places? I am not sure how it works for the attached lock file in the discussion. Can we add a unit test with that file?